### PR TITLE
Skip time zone in test by using Calendar, not Date

### PR DESCRIPTION
**Pull request by mike-burns**
A `java.util.Date` has the local time zone embedded in it, so
instantiating it with the seconds since the epoch will always produce a
local time that varies.

Conversely, `java.util.Calendar` works differently; it always gives back
a date (using `Calendar.getTime()`) that matches the numbers passed in,
even if the time zone does not match. We can use this to ignore the time
zone component of the date but get a `Date` instance for the exact
desired date and time.

### CODE REVIEW

**Comment by mike-burns** [comment]
I think there's some code duplication in the tests. How about a factory method for generating these mock conditions, shared between all the test files? Is that reasonable? I'm about to look at that.

File: app/src/test/kotlin/com/thoughtbot/tropos/viewmodels/WeatherToolbarViewModelTest.kt [hunk 1]
```diff
@@ -25,8 +25,9 @@ class WeatherToolbarViewModelTest() {
   lateinit var context: Context
 
   val mockCondition: Condition = {
-    val timeStamp = 1484180189 * 1000L // equivalent to Wed Jan 11 16:16:29 PST 2017
-    val date = Date(timeStamp)
+    val cal = Calendar.getInstance()
+    cal.set(2017, 1, 11, 16, 16, 29)
```
Comment (mike-burns) [inline app/src/test/kotlin/com/thoughtbot/tropos/viewmodels/WeatherToolbarViewModelTest.kt]: That's February 11th, not January. Read the docs, Mike!

**Comment by mike-burns** [comment]
How's this look? Is `.getInstance()` a reasonable name? Is there a better approach here? I'm not too familiar with the conventions 'round these parts.

**Comment by mandybess** [comment]
@mike-burns factory method is totally reasonable. Curious about changing from `Date` to `Calendar` - those are different types, do the tests still pass / compile? I would imagine since a `Condition` object is expecting a `Date` parameter, it wouldn't even build?

File: app/src/test/kotlin/com/thoughtbot/tropos/testUtils/MockCondition.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import android.location.Location
+import com.thoughtbot.tropos.data.Condition
+import com.thoughtbot.tropos.data.Icon
+import com.thoughtbot.tropos.data.Unit
+import com.thoughtbot.tropos.data.WindDirection
+import java.util.*
+
+
+class MockCondition {
```
Comment (mandybess) [inline app/src/test/kotlin/com/thoughtbot/tropos/testUtils/MockCondition.kt]: I realize in the tests i called the variables `mockCondition` but we aren't mocking here, rather we are faking so maybe change to `Fake`? Or we maybe we should wait to do that all at once in a later PR?

I'm also wondering if there is a way to create package private extension functions - which would eliminate the need for the `companion object`. And we could just have `Condition.createFake()`? I'm not sure if that's a great signature though? Just thinking out loud here in word form ha

Comment (mike-burns) [reply app/src/test/kotlin/com/thoughtbot/tropos/testUtils/MockCondition.kt]: Sure, I don't mind renaming "mock" to "fake".

`Condition.createFake()` would have to be an extension on the companion object, right? Something like:

```kotlin
fun Condition.Companion.createFake() {
  val date = TODO()
  // ...
  return Condition(...)
}
```

I haven't explored `internal` visibility yet -- I wonder if that will help us scope such an extension to only the test suite.

Comment (mike-burns) [reply app/src/test/kotlin/com/thoughtbot/tropos/testUtils/MockCondition.kt]: Actually, what about exporting a `fakeCondition()` function from `com.thoughtbot.tropos.testUtils`? Gonna try that.

**Comment by mike-burns** [comment]
Oh, to answer the `Date`/those are different types question: I call `Calendar.getTime()` to get a `Date` instance.

**Comment by mike-burns** [comment]
(I'll rebase and clean up the commits after all the code looks good.)

**Comment by mike-burns** [comment]
I had forgotten about this. Should I rebase,clean up the commits, and merge?

**Comment by mandybess** [comment]
@mike-burns yes! i too had forgotten about this! but feel free to clean up and merge!

**Pull request merged**